pkg/deploy/kubernetes_image_puller.go

```diff
@@ -379,6 +385,77 @@ func UpdateImagePullerSpecIfEmpty(ctx *DeployContext) (orgv1.CheClusterSpecImage
 	return ctx.CheCluster.Spec.ImagePuller, nil
 }
 
+func AddRelatedImages(ctx *DeployContext) {
+	relatedImages := []string{}
+	for _, env := range os.Environ() {
+		if !strings.HasPrefix(env, "RELATED_IMAGE_") {
+			continue
+		}
+
+		keyValue := strings.Split(env, "=")
+		key := keyValue[0]
+		value := keyValue[1]
+
+		if ShouldAddImage(key) {
+			key = key[len("RELATED_IMAGE_"):]
+			key, err := ConvertToRFC1123(key)
+			if err != nil {
+				logrus.Errorf(err.Error())
+				continue
+			}
+			relatedImages = append(relatedImages, key+"="+value)
+		}
+	}
+
+	if len(ctx.CheCluster.Spec.ImagePuller.Spec.Images) != 0 &&
+		!strings.HasSuffix(ctx.CheCluster.Spec.ImagePuller.Spec.Images, ";") {
+		ctx.CheCluster.Spec.ImagePuller.Spec.Images += ";"
+	}
+
+	ctx.CheCluster.Spec.ImagePuller.Spec.Images += strings.Join(relatedImages[:], ";")
+	ctx.ClusterAPI.Client.Update(context.TODO(), ctx.CheCluster, &client.UpdateOptions{})
+}
+
+func ShouldAddImage(key string) bool {
```

Comment:
It is not good to have this list hardcoded, because it won't work for CRW [1] since
1. they are named like `RELATED_IMAGE_codeready_workspaces_`.
2. CRW mighty need another list of images

[1] https://github.com/redhat-developer/codeready-workspaces-operator/blob/crw-2.9-rhel-8/manifests/codeready-workspaces.csv.yaml

Reply:
Why not put them directly into a sample [1] which is used for a new Eclipse Che deployment ?
[1] https://github.com/eclipse-che/che-operator/blob/main/deploy/crds/org_v1_che_cr.yaml#L11

Reply:
Thank you for the review

> Why not put them directly into a sample [1] which is used for a new Eclipse Che deployment ?
[1] https://github.com/eclipse-che/che-operator/blob/main/deploy/crds/org_v1_che_cr.yaml#L11

If I understand correctly, I should consider defining the image prefixes in [1] and read them from there, since these prefixes are che-specific?

Reply:
1. Not the prefixes but images directly into a CR during releasing. It requires modifying bash script instead of go code.
For instance we can run script to modify CR and CSV files here [1]

```yaml
imagePuller:
  spec:
    images:  <...>
```

2. For CRW we can override those list by modifying sync scripts [2] [3] 

[1] https://github.com/eclipse-che/che-operator/blob/main/make-release.sh#L311
[2] https://github.com/redhat-developer/codeready-workspaces-operator/blob/crw-2-rhel-8/build/scripts/sync-che-operator-to-crw-operator.sh
[3] https://github.com/redhat-developer/codeready-workspaces-operator/blob/crw-2-rhel-8/build/scripts/sync-che-olm-to-crw-olm.sh

Reply:
@tolusha am I correct that related images in csv would only be available for released versions of the project / product?

Reply:
Right.
We add them to csv during release process

https://github.com/eclipse-che/che-operator/blob/main/make-release.sh#L225
https://github.com/eclipse-che/che-operator/blob/main/olm/release-olm-files.sh#L122-L125